### PR TITLE
test: add unit tests for detect distro

### PR DESCRIPTION
**Pull request by phillebaba**
## Description

Adds tests to detect distro logic.

## Related Issue

Relates to #2512

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/.github/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/pkg/cluster/distro_test.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+					},
+				},
+			},
+		},
+		{
+			distro: DistroIsK3s,
+			node: corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"": "node.kubernetes.io/instance-type=k3s",
```
Comment (phillebaba) [inline src/pkg/cluster/distro_test.go:53]: This passes but I think the logic is actually wrong.

File: src/pkg/cluster/distro_test.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+					ProviderID: "kind://docker/kind/kind-control-plane",
+				},
+			},
+		},
+		{
+			distro: DistroIsMicroK8s,
+			node: corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"": "microk8s.io/cluster=true",
```
Comment (phillebaba) [inline src/pkg/cluster/distro_test.go:79]: This passes but I think the logic is actually wrong.

File: src/pkg/cluster/distro.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		return DistroIsGKE
+	}
+
+	// https://github.com/kubernetes/kubernetes/blob/v1.23.4/staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go#L46
+	if aksNodeRegex.MatchString(node.Spec.ProviderID) {
+		return DistroIsAKS
+	}
+
+	labels := node.GetLabels()
+	for _, label := range labels {
```
Comment (phillebaba) [inline src/pkg/cluster/distro.go:66]: This looks wrong. It assumes the label is in selector format. In reality is should look at key and value instead.

**Review by Noxsios — COMMENTED** [review]
LGTM, just needs some merge nonsense cleaned up from (some stuff referenced no longer exists in the `k8s` package), running `make build` should show the offending imports.

**Review by AustinAbro321 — APPROVED** [review]

**Pull request merged**